pkg/cmd/server/kubernetes/master.go

```diff
@@ -69,11 +71,42 @@ const (
 // a single string value).
 func (c *MasterConfig) InstallAPI(container *restful.Container) ([]string, error) {
 	c.Master.RestfulContainer = container
-	_, err := master.New(c.Master)
+
+	// keep the old value
+	enableCoreControllers := c.Master.EnableCoreControllers
+
+	// explicitly set to false because master.New will start the bootstrap controller otherwise, and
+	// we want to start it ourselves so we can possibly override the EndpointReconciler
+	c.Master.EnableCoreControllers = false
+
+	m, err := master.New(c.Master)
 	if err != nil {
 		return nil, err
 	}
 
+	// restore the original value
+	c.Master.EnableCoreControllers = enableCoreControllers
```

Comment:
Yeah, I see that now. Fortunately for the moment, that value is only ever used in 1 place, to run the bootstrap controller. And the refactoring upstream that you want will make this a non-issue going forward. We could probably just remove this line for now.